np.histogram erroneously emit warning when density=False

As seen in https://travis-ci.org/ejeschke/ginga/jobs/395720074 . The actual code in Ginga calling histogram is:
```python
# get image histogram
hist, bins = numpy.histogram(some_array, hashsize, density=False)
```
Note that `normed` was not used. However, it caused this warning in Numpy ~~dev~~ prerelease:
```python
        else:
            if normed is not None:
                # 2018-06-13, numpy 1.15.0 (this was not noisily deprecated in 1.6)
                warnings.warn(
                        "Passing normed=False is deprecated, and has no effect. "
                        "Consider passing the density argument instead.",
>                       DeprecationWarning, stacklevel=2)
E               DeprecationWarning: Passing normed=False is deprecated, and has no effect. Consider passing the density argument instead.
```

I think this is caused by #11352 . When `density=False`, it goes into the logic block that forces `normed=False`, which resulted in it going into this logic block in `np.histogram()`:
```python
else:
    if normed is not None:
        ...
```

cc @ejeschke

--- a/numpy/lib/histograms.py
+++ b/numpy/lib/histograms.py
@@ -782,7 +782,7 @@ def histogram(a, bins=10, range=None, normed=None, weights=None,
                     "The normed argument is ignored when density is provided. "
                     "In future passing both will result in an error.",
                     DeprecationWarning, stacklevel=2)
-        normed = False
+        normed = None
 
     if density:
         db = np.array(np.diff(bin_edges), float)

--- a/numpy/lib/tests/test_histograms.py
+++ b/numpy/lib/tests/test_histograms.py
@@ -78,6 +78,10 @@ class TestHistogram(object):
         assert_array_equal(a, .1)
         assert_equal(np.sum(a * np.diff(b)), 1)
 
+        # Test that passing False works too
+        a, b = histogram(v, bins, density=False)
+        assert_array_equal(a, [1, 2, 3, 4])
+
         # Variale bin widths are especially useful to deal with
         # infinities.
         v = np.arange(10)

